### PR TITLE
Adding missing QJIT tests for measurement processes

### PR DESCRIPTION
**Pull request by willjmax**
Adding tests marked with `xfail` for measurement processes not supported by Catalyst. This PR addresses [sc-72627].

### CODE REVIEW

**Review by dime10 — APPROVED** [review]
Thank you @willjmax! Just make sure to run configure black on all your python files :)

File: frontend/test/pytest/test_measurements_results.py
Comment (soranjh) [inline frontend/test/pytest/test_measurements_results.py:1]: Are these tests checked locally with PL (not Catalyst) to make sure that they pass? For instance, to make sure that the asserts have no bug and don't check something incorrect and also to make sure the reference values are correct.

File: frontend/test/pytest/test_measurements_shots_results.py [hunk 1]
```diff
@@ -481,6 +481,109 @@ def circuit(theta):
         assert np.allclose(result, expected, atol=tol_stochastic, rtol=tol_stochastic)
 
 
+class TestVnEntropy:
```
Comment (soranjh) [inline frontend/test/pytest/test_measurements_shots_results.py:484]: What is the difference between this test and the one in test_measurements_results.py? I assume the main difference is in setting `shots` in the device? 

If this is correct, are such measurements supported with finite shots on default.qubit?

**Review by soranjh — COMMENTED** [review]
Thanks @willjmax, good from my side after we address my questions.

**Comment by paul0403** [comment]
@willjmax Looks good!

I know github says no conflicts but can we do a quick pull of main to be absolutely certain (and check code coverage)?

**Review by soranjh — APPROVED** [review]
Thanks @willjmax, please don't forget to i) update the changelog and ii) put a link to the shortcut story.

**Pull request merged**